kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/InputFormatReader.java

```diff
@@ -164,14 +164,14 @@ private boolean advance() {
       }
     } catch (RuntimeException e) {
       this.state = ReaderWriterState.ERROR;
-      throw new DatasetReaderException("Cannot advance reader", e);
+      throw new DatasetOperationException("Cannot advance reader", e);
     } catch (IOException e) {
       this.state = ReaderWriterState.ERROR;
       throw new DatasetIOException("Cannot advance reader", e);
     } catch (InterruptedException e) {
       // error: it is unclear whether the underlying reader is valid
       this.state = ReaderWriterState.ERROR;
-      throw new DatasetReaderException("Interrupted", e);
+      throw new DatasetOperationException("Interrupted", e);
```

Comment:
This should call `Thread.currentThread().interrupt()` rather than swallowing the interrupt.

http://www.ibm.com/developerworks/library/j-jtp05236/